tests/test_functions.py

```diff
@@ -223,3 +226,31 @@ def cleanup_check():
     Checks to make sure all endpoints were tested."""
 
     test_state.cleanup_check()
+
+
+def run_bash(command):
+    process = subprocess.Popen(command.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+
+def file_tail(filename, length):
```

Comment:
Why not read the whole file into memory and then print the lines? failure in os.popen can be hard to handle

Reply:
Good idea- I was probably over-optimizing